File: LoveMap/src/main/java/com/example/lovemap/model/entity/Anniversary.java
package com.example.lovemap.model.entity;

import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 纪念日实体类
 * 对应数据库表：anniversary
 */
@Data
public class Anniversary {

    /**
     * 主键
     */
    private Long id;

    /**
     * 所属群组ID（情侣共享；未绑定情侣时为 null）
     */
    private Long groupId;

    /**
     * 创建人用户ID（仅未绑定情侣的个人纪念日使用）
     */
    private Long userId;

    /**
     * 纪念日标题
     */
    private String name;

    /**
     * 纪念日日期
     */
    private LocalDate anniversaryDate;

    /**
     * 是否每年重复提醒
     */
    private Boolean isRecurring;

    /**
     * 提醒提前天数
     */
    private Integer remindDays;

    /**
     * 备注描述
     */
    private String description;

    /**
     * 创建时间
     */
    private LocalDateTime createdAt;

    /**
     * 更新时间
     */
    private LocalDateTime updatedAt;
}

File: loveofus-frontend/src/api/anniversary.ts
import request from '@/utils/request'

/**
 * 纪念日数据类型
 */
export interface Anniversary {
  id: number
  name: string
  anniversaryDate: string
  isRecurring: boolean
  remindDays: number
  description: string
  daysUntil?: number
  createdAt: string
  updatedAt: string
}

/**
 * 创建/更新纪念日请求参数
 */
export interface AnniversaryParams {
  name: string
  anniversaryDate: string
  isRecurring?: boolean
  remindDays?: number
  description?: string
}

/**
 * 获取纪念日列表
 */
export function getAnniversaryList() {
  return request.get<Anniversary[]>('/anniversaries')
}

/**
 * 获取纪念日详情
 * @param id 纪念日ID
 */
export function getAnniversary(id: number) {
  return request.get<Anniversary>(`/anniversaries/${id}`)
}

/**
 * 创建纪念日
 * @param data 纪念日数据
 */
export function createAnniversary(data: AnniversaryParams) {
  return request.post<Anniversary>('/anniversaries', data)
}

/**
 * 更新纪念日
 * @param id 纪念日ID
 * @param data 纪念日数据
 */
export function updateAnniversary(id: number, data: AnniversaryParams) {
  return request.put<Anniversary>(`/anniversaries/${id}`, data)
}

/**
 * 删除纪念日
 * @param id 纪念日ID
 */
export function deleteAnniversary(id: number) {
  return request.delete<void>(`/anniversaries/${id}`)
}
